Applications/spire/source/charting/chart_view.cpp

```diff
@@ -117,9 +116,10 @@ void ChartView::set_region(const ChartPoint& top_left,
   m_y_range = m_top_left.m_y - m_bottom_right.m_y;
   m_y_axis_step = calculate_step(m_model->get_y_axis_type(), m_y_range);
   update_origins();
-  m_loaded_data = m_model->load(m_top_left.m_x, m_bottom_right.m_x);
-  m_loaded_data.then([&] (auto result) {
-    m_candlesticks = result.Get();
+  m_loaded_candlestick_promise = m_model->load(m_top_left.m_x,
+    m_bottom_right.m_x);
+  m_loaded_candlestick_promise.then([&] (auto result) {
+    m_candlesticks = std::move(result.Get());
```

Comment:
Line 122 can get called at any arbitrary point in the future, imagine it gets called 5 years from now... is the only operation you want to perform a re-assignment?

Reply:
When I ask an ambiguous question it's mostly because I want you to think it through carefully. It's a good opportunity for you to gain a better understanding as opposed to me just giving you the answer to every detail. There is an important concept that is worth gaining an intuition for.